### PR TITLE
Feature/player surrender

### PR DESCRIPTION
**Pull request by zhihdd**
# Description

# How to test

# New Dependencies

# TODO
1. data-service 整合
2. adapter layer / ORM
3. server -> ORM -> mongodb

### CODE REVIEW

**Review by allenh — CHANGES_REQUESTED** [review]
* Please update `README.md` with the vocabulary translation
* Do we want unit tests and are we enforcing a minimum unit test coverage?
* What's the idea hind having files that are getting implemented by their respective implementation classes living under `/usecases` and not with their implementation class folder?  Eg: why is `repositoy.abstract.ts` under `/usecase` and not under `/repository`?

If the idea is that `usecase` is supposed to include all the business logic then would it make sense to rename `usecase` to `service` and group all the use case methods under their logical business logical name? For example `surrender` and `furigoma` use-cases could both be under `GungiService`:
```ts
class GungiService {
  async furigoma() {}
  async surrender() {}
}
```

File: server/package.json [hunk 1]
```diff
@@ -18,12 +18,17 @@
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json"
+    "test:e2e": "jest --config ./test/jest-e2e.json ",
+    "test:unit": "jest --config ./test/unitTest/jest.confg.js",
+    "test:all": "npm run test:unit && npm run test:e2e"
```
[the remark this answers is not in the document]
Comment (zhihdd) [reply server/package.json:23]: Fix by closing database connection after alll test

File: server/src/usecases/repository.abstract.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default abstract class IRepository<T> {
```
[the remark this answers is not in the document]
Comment (zhihdd) [reply server/src/usecases/repository.abstract.ts:1]: I thougt interface can not be used by IoC container. But, after couple trying, it is work.

**Review by allenh — COMMENTED** [review]
I think overall this PR is in a good spot to merge. There are still some minor issues that need fixing but in the interest of time and progress I don't think these issues warrant a merge block and these issues can be fixed later on. If everyone is ok with how the project looks and after #14 is merged to unblock the pipeline then we can merge this to master.

**Review by noracami — APPROVED** [review]
LGTM

**Pull request merged**